### ISSUE TITLE
Support the STORAGE_EMULATOR_HOST environment variable for StorageClient

### ISSUE THREAD

**Issue by EricStG** [googleapis/google-cloud-dotnet#11522]
**Is your feature request related to a problem? Please describe.**
We're using the firebase emulator, and we want to be able to configure the StorageClient via environment variables

 **Describe the solution you'd like**
We'd like to be able to configure the StorageClient's host via the STORAGE_EMULATOR_HOST environment variable

 **Describe alternatives you've considered**
We are currently setting the address via code, as described [here](https://github.com/hemanshv/dotnet-docs-samples/blob/main/storage/api/Storage.Samples.Tests/SetClientEndpointTest.cs).

 **Additional context**
This is already supported by the [GO](https://cloud.google.com/go/docs/reference/cloud.google.com/go/storage/latest) and [Node](https://github.com/googleapis/nodejs-storage/issues/2069) clients

**Comment by jskeet** [comment]
Will have a look at this. One thing that the .NET libraries are consistent about is not *just* using an environment variable. We'd do this in the same way - see https://cloud.google.com/dotnet/docs/reference/help/emulators for examples.

(I wasn't aware that the Firebase emulator implemented a full Storage service... Will need to test it thoroughly.)

**Comment by jskeet** [comment]
Hmmm... I've just had made a first attempt at this, and the emulator just rejects all requests. I need to look in more detail as to what it *will* accept.

**Comment by jskeet** [comment]
Ah - my mistake, I'd used the emulator UI port instead of the API port. Doh. Looks like it's working now - with the limitations around no bucket operations, etc. Will see if I can get it more production-ready and create a PR.

**Comment by EricStG** [comment]
We have it working here without any changes, at least for basic upload/download
Except for a workaround on UpdateObjectAsync, which fails (we haven't dug too much into it for our needs though)

**Comment by jskeet** [comment]
Will look into UpdateObjectAsync as part of the work - thanks for calling that out.

**Comment by EricStG** [comment]
We were having errors trying to update both `CacheControl` and `Metadata` with `UpdateObjectAsync`.
The error was:
```
The service storage has thrown an exception.
HttpStatusCode is NotImplemented.
No JSON error details were specified.
Raw error details are: Not Implemented
```
Could be a limitation on the emulator, but it would not be [documented](https://firebase.google.com/docs/emulator-suite/connect_storage#differences_from_google_cloud_storage).

**Comment by jskeet** [comment]
Yes, I suspect it's an emulator limitation - as you say, object update claims to be supported, but I get the same results as you, even changing a different field (customDateTime). I can *patch* a change to metadata, but not a change to customDateTime. I don't know how the Firebase Emulator versioning keeps up with the Storage API versioning. I'll ask internally about the object update part not working, but obviously that's somewhat separate to the client library support itself.

**Comment by jskeet** [comment]
This is implemented in #11535, which will be in the next release.
